CMakeLists.txt

```diff
@@ -337,6 +337,26 @@ endif (WIN32)
 
 endif (OVERRIDE_CMAKE_CXX_FLAGS)
 
+OPTION(CMAKE_ENABLE_CODE_COVERAGE "Enable code coverage" OFF)
+
+if(CMAKE_ENABLE_CODE_COVERAGE)
+
+  string(TOUPPER ${CMAKE_BUILD_TYPE} UPPERCASE_CMAKE_BUILD_TYPE)
+  if(NOT UPPERCASE_CMAKE_BUILD_TYPE STREQUAL DEBUG)
+    message( WARNING "Code coverage results with an optimised (non-Debug) build may be misleading" )
+  endif(NOT UPPERCASE_CMAKE_BUILD_TYPE STREQUAL DEBUG)
+
+  if(CLR_CMAKE_PLATFORM_UNIX)
+    add_definitions(-fprofile-arcs -ftest-coverage -fPIC)
```

Comment:
@janvorli I think the -fPIC is not getting added all the libraries. I get this below error when I compile without it from the root CMakeLists.txt

*/usr/bin/ld: ../../../inc/libcorguids.a(cordebug_i.c.o): relocation R_X86_64_PC32 against undefined symbol `llvm_gcda_start_file' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
clang: error: linker command failed with exit code 1 (use -v to see invocation)*